### DISCUSSION TITLE
Sensor Fusion Testing and fixing

### DISCUSSION THREAD

**Discussion by aburt2** [IDMIL/T-Stick#41] [Ideas]
Let 2024 be the year we properly evaluate test and fix the sensor fusion on the T-Stick and finally close the [sensor fusion issue](https://github.com/IDMIL/T-Stick/issues/7).

From what I recall from previous discussions this was what still needed to be done:
- [ ] Setup a Calibration routine
- [ ] Test and evaluate the different sensor fusion algorithms and implementations

**Comment by aburt2** [comment]
Pinging the relevant people @docsunset, @edumeneses , @malloch, @bboettcher3

**Reply by edumeneses** [reply]
Ok, All those functions live in [puara-gestures](https://github.com/Puara/puara-gestures) right now. I started revising the whole repo but took a break for the holidays.

I suggest a full refactoring to the puara-gesture library as it doesn't play well outside ESP32 right now (it calls some espressif-specific functions).

My plan is to change the structure, have a common namespace, and create classes for each gesture/input data. 

I also plan to keep both @DocSunset and @malloch's orientation implementations and have, as @aburt2 suggested, proper calibration routines for them.

Do you all want to trace a plan (with schedule and deadlines)? We can meet in the second or third week of January 2024 to set the dates as tasks. Please let me know in this thread or via Slack.

**Reply by aburt2** [reply]
Setting a meeting to organise deadlines sounds good. Third week of January works for me
